### ISSUE TITLE
tests fail on 32bit architectures

### ISSUE THREAD

**Issue by jelmer** [heimdal/heimdal#175]
WIth current master, a couple of tests are failing on 32bit architectures only:

FAIL: check-kdc
FAIL: check-kdc-weak
FAIL: check-kpasswdd
FAIL: check-referral
FAIL: check-tester
FAIL: check-uu 

all 64 bit architectures seem to be fine. Unfortunately I don't have the actual logs here.

**Comment by nicowilliams** [comment]
Thanks, I'll take a look.

**Comment by jelmer** [comment]
Ping, any news on this?

**Comment by madscientist159** [comment]
Still running into this on Debian

**Comment by madscientist159** [comment]
On ARM64:

```
==============================================
   Heimdal 1.6.99: tests/kdc/test-suite.log
==============================================

# TOTAL: 18
# PASS:  13
# SKIP:  3
# XFAIL: 0
# FAIL:  2
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

SKIP: check-hdb-mitdb
=====================

no MIT KDB support
SKIP check-hdb-mitdb (exit status: 77)

FAIL: check-kdc
===============

Creating database
Check parser
Doing database check
Extracting enctypes
deleting all but des enctypes on kt-des3 in keytab
checking globbing keys rules
Starting kdc
Waiting for KDC to start, looking logfile messages.log
Have waited 2 seconds
Starting kpasswdd
Getting client initial tickets with wrong password
Getting client initial tickets
Doing krbtgt key rollover
Getting tickets
Listing tickets
Getting client initial tickets (http transport)
Testing capaths logic
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST2.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST3.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST4.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM5.FR
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM6.US
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM7.UK
Testing capaths logic (reverse order)
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST4.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST3.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST2.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM7.UK
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM6.US
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM5.FR
Testing forwardable/renewable flag copying in TGS-REQ
Specific enctype
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets
Getting client initial tickets
Getting tickets (aes256-cts-hmac-sha1-96)
Getting client initial tickets for cross realm case
Getting cross realm tickets (aes256-cts-hmac-sha1-96)
  checking we we got back right ticket
  checking if ticket is useful
Trying x-realm TGT with kvno 0 case
Getting cross realm tickets
Getting service ticket
Trying x-realm TGT with kvno 0 case with key rollover
Getting cross realm tickets
Rolling over cross realm keys
Getting service ticket
Start tracing kdc, then hit return
Trying x-realm TGT with no kvno case
Getting cross realm tickets
Getting service ticket
Trying x-realm TGT with no kvno case with key rollover
Getting cross realm tickets
Rolling over cross realm keys
Getting service ticket
Start tracing kdc, then hit return
try all permutations
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96)
Getting client initial tickets ip based name
Getting ip based name tickets
  checking we we got back right ticket
  checking if ticket is useful
Getting client initial tickets ip based name (alias)
Getting ip based name tickets (alias) host/ip.test.h5l.org
  checking we we got back right ticket
  checking if ticket is useful
Getting ip based name tickets (alias) host/10.11.12.14
  checking we we got back right ticket
  checking if ticket is useful
Getting server initial tickets
Listing tickets
Getting key for key that are a subset in keytab compared to kdb
initial tickets for deleted user test case
try getting ticket with deleted user
cross realm case (deleted user)
rename user
rename user to another realm
deleting all but aes enctypes on krbtgt
deleting all but des enctypes on server-des3
try all permutations (only aes)
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96) (server des3 only)
deleting all enctypes on krbtgt
try initial ticket w/o and keys on krbtgt
adding random aes key
try initial ticket with random aes key on krbtgt
try anonymous pkinit
Trying pk-init (principal in certificate) 
Trying pk-init (principal in pki-mapping) 
Trying pk-init (password protected key) 
Trying pk-init (proxy cert) 
Trying pk-init (principal in certificate) --pk-use-enckey
Trying pk-init (principal in pki-mapping) --pk-use-enckey
Trying pk-init (password protected key) --pk-use-enckey
Trying pk-init (proxy cert) --pk-use-enckey
tickets for impersonate test case
  negative check
test constrained delegation
  try using the credential
  negative check
test constrained delegation impersonation (non forward)
test constrained delegation impersonation (missing KRB5SignedPath)
kinit -R
test_renew
checking server aliases
Getting tickets
   verify entry in keytab
   verify entry in keytab with any
   verify failure with alias entry
   verify alias entry in keytab with any
testing removal of keytab
Checking client pw expire
   kinit passes
   test_gic passes
Checking password expiration
Checking client pw expire
test failed
./check-kdc: 1: kill: No such process

signal killing kdc kpasswdd
FAIL check-kdc (exit status: 1)

FAIL: check-kdc-weak
====================

Creating database
Check parser
Doing database check
Extracting enctypes
deleting all but des enctypes on kt-des3 in keytab
checking globbing keys rules
Starting kdc
Waiting for KDC to start, looking logfile messages.log
Have waited 2 seconds
Starting kpasswdd
Getting client initial tickets with wrong password
Getting client initial tickets
Doing krbtgt key rollover
Getting tickets
Listing tickets
Getting client initial tickets (http transport)
Testing capaths logic
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST2.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST3.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST4.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM5.FR
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM6.US
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM7.UK
Testing capaths logic (reverse order)
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST4.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST3.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST2.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM7.UK
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM6.US
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM5.FR
Testing forwardable/renewable flag copying in TGS-REQ
Specific enctype
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets
Getting client initial tickets
Getting tickets (aes256-cts-hmac-sha1-96)
Getting client initial tickets for cross realm case
Getting cross realm tickets (aes256-cts-hmac-sha1-96)
  checking we we got back right ticket
  checking if ticket is useful
Trying x-realm TGT with kvno 0 case
Getting cross realm tickets
Getting service ticket
Trying x-realm TGT with kvno 0 case with key rollover
Getting cross realm tickets
Rolling over cross realm keys
Getting service ticket
Start tracing kdc, then hit return
Trying x-realm TGT with no kvno case
Getting cross realm tickets
Getting service ticket
Trying x-realm TGT with no kvno case with key rollover
Getting cross realm tickets
Rolling over cross realm keys
Getting service ticket
Start tracing kdc, then hit return
try all permutations
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96)
Getting client initial tickets ip based name
Getting ip based name tickets
  checking we we got back right ticket
  checking if ticket is useful
Getting client initial tickets ip based name (alias)
Getting ip based name tickets (alias) host/ip.test.h5l.org
  checking we we got back right ticket
  checking if ticket is useful
Getting ip based name tickets (alias) host/10.11.12.14
  checking we we got back right ticket
  checking if ticket is useful
Getting server initial tickets
Listing tickets
Getting key for key that are a subset in keytab compared to kdb
initial tickets for deleted user test case
try getting ticket with deleted user
cross realm case (deleted user)
rename user
rename user to another realm
deleting all but aes enctypes on krbtgt
deleting all but des enctypes on server-des3
try all permutations (only aes)
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96) (server des3 only)
deleting all enctypes on krbtgt
try initial ticket w/o and keys on krbtgt
adding random aes key
try initial ticket with random aes key on krbtgt
try anonymous pkinit
Trying pk-init (principal in certificate) 
Trying pk-init (principal in pki-mapping) 
Trying pk-init (password protected key) 
Trying pk-init (proxy cert) 
Trying pk-init (principal in certificate) --pk-use-enckey
Trying pk-init (principal in pki-mapping) --pk-use-enckey
Trying pk-init (password protected key) --pk-use-enckey
Trying pk-init (proxy cert) --pk-use-enckey
tickets for impersonate test case
  negative check
test constrained delegation
  try using the credential
  negative check
test constrained delegation impersonation (non forward)
test constrained delegation impersonation (missing KRB5SignedPath)
kinit -R
test_renew
checking server aliases
Getting tickets
   verify entry in keytab
   verify entry in keytab with any
   verify failure with alias entry
   verify alias entry in keytab with any
testing removal of keytab
Checking client pw expire
   kinit passes
   test_gic passes
Checking password expiration
Checking client pw expire
test failed
../../tests/kdc/check-kdc: 1: kill: No such process

signal killing kdc kpasswdd
FAIL check-kdc-weak (exit status: 1)

SKIP: check-kpasswdd
====================

Creating database for TEST.H5L.SE
Creating database for TEST2.H5L.SE
Doing database check for TEST.H5L.SE TEST2.H5L.SE
Starting kdc
Waiting for KDC to start, looking logfile messages.log
Have waited 2 seconds
Starting kpasswdd
Getting client initial tickets
Getting tickets
Listing tickets
checking TEST.H5L.SE
Changing password
test failed
2016-09-22T03:12:39 error message: Did not find a plugin for ccache_ops: 2
2016-09-22T03:12:39 error message: Did not find a plugin for ccache_ops: 2
2016-09-22T03:12:39 krb5_get_creds: requesting a ticket for host/datan.test.h5l.se@TEST.H5L.SE
2016-09-22T03:12:39 error message: Name canon failed while searching keytab: 0
2016-09-22T03:12:39 error message: No authorization data: 2
2016-09-22T03:12:39 error message: Ticket have not authorization data of type 128: 2
2016-09-22T03:12:39 krb5_get_creds: requesting a ticket for host/datan.test.h5l.se@TEST.H5L.SE
2016-09-22T03:12:39 error message: Name canon failed while searching keytab: 0
2016-09-22T03:12:39 error message: No authorization data: 2
2016-09-22T03:12:39 error message: Ticket have not authorization data of type 128: 2
2016-09-22T03:12:40 error message: Did not find a plugin for ccache_ops: 2
./check-kpasswdd: 1: kill: No such process

signal killing kdc
SKIP check-kpasswdd (exit status: 77)

SKIP: check-iprop
=================

SKIP check-iprop (exit status: 77)
```

**Comment by nacc** [comment]
On i386/armhf/powerpc, Ubuntu 16.10, we're only seeing two testcase fail. Reproduced locally on i386:

```
FAIL: check-kdc
Creating database
Check parser
Doing database check
Extracting enctypes
deleting all but des enctypes on kt-des3 in keytab
checking globbing keys rules
Starting kdc
Waiting for KDC to start, looking logfile messages.log
Have waited 2 seconds
Starting kpasswdd
Getting client initial tickets with wrong password
Getting client initial tickets
Doing krbtgt key rollover
Getting tickets
Listing tickets
Getting client initial tickets (http transport)
Testing capaths logic
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST2.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST3.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST4.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM5.FR
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM6.US
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM7.UK
Testing capaths logic (reverse order)
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST4.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST3.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> TEST2.H5L.SE
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM7.UK
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM6.US
Getting x-realm tickets with capaths for TEST.H5L.SE -> SOME-REALM5.FR
Testing forwardable/renewable flag copying in TGS-REQ
Specific enctype
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets
Getting client initial tickets
Getting tickets (aes256-cts-hmac-sha1-96)
Getting client initial tickets for cross realm case
Getting cross realm tickets (aes256-cts-hmac-sha1-96)
  checking we we got back right ticket
  checking if ticket is useful
Trying x-realm TGT with kvno 0 case
Getting cross realm tickets
Getting service ticket
Trying x-realm TGT with kvno 0 case with key rollover
Getting cross realm tickets
Rolling over cross realm keys
Getting service ticket
Start tracing kdc, then hit return
Trying x-realm TGT with no kvno case
Getting cross realm tickets
Getting service ticket
Trying x-realm TGT with no kvno case with key rollover
Getting cross realm tickets
Rolling over cross realm keys
Getting service ticket
Start tracing kdc, then hit return
try all permutations
Getting client initial tickets (aes256-cts-hmac-sha1-96)
Getting tickets (aes256-cts-hmac-sha1-96 ->  aes256-cts-hmac-sha1-96)
Getting client initial tickets ip based name
Getting ip based name tickets
  checking we we got back right ticket
  checking if ticket is useful
Getting client initial tickets ip based name (alias)
Getting ip based name tickets (alias) host/ip.test.h5l.org
  checking we we got back right ticket
  checking if ticket is useful
Getting ip based name tickets (alias) host/10.11.12.14
  checking we we got back right ticket
  checking if ticket is useful
Getting server initial tickets
Listing tickets
Getting key for key that are a subset in keytab compared to kdb
initial tickets for deleted user test case
lt-kadmin: del remove@TEST.H5L.SE: Input/output error
test failed
2016-09-29T10:20:54 error message: Did not find a plugin for ccache_ops: 2
2016-09-29T10:20:54 error message: encryption type des-cbc-crc is disabled: -1765328234
2016-09-29T10:20:54 krb5_get_init_creds: loop 1
2016-09-29T10:20:54 KDC send 0 patypes
2016-09-29T10:20:54 error message: Did not find a plugin for send_to_kdc: 2
2016-09-29T10:20:54 Trying to find service kdc for realm TEST.H5L.SE flags 0
2016-09-29T10:20:54 error message: Did not find a plugin for service_locator: 2
2016-09-29T10:20:54 configuration file for realm TEST.H5L.SE found
2016-09-29T10:20:54 submissing new requests to new host
2016-09-29T10:20:54 error message: Did not find a plugin for send_to_kdc: 2
2016-09-29T10:20:54 connecting to host: udp ::1:49188 (localhost) tid: 00000001
2016-09-29T10:20:54 Queuing host in future (in 3s), its the 2 address on the same name: udp 127.0.0.1:49188 (localhost) tid: 00000002
2016-09-29T10:20:54 writing packet: udp ::1:49188 (localhost) tid: 00000001
2016-09-29T10:20:54 AS-REQ remove@TEST.H5L.SE from IPv6:::1 for krbtgt/TEST.H5L.SE@TEST.H5L.SE
2016-09-29T10:20:54 Client sent patypes: REQ-ENC-PA-REP
2016-09-29T10:20:54 Looking for PK-INIT(ietf) pa-data -- remove@TEST.H5L.SE
2016-09-29T10:20:54 Looking for PK-INIT(win2k) pa-data -- remove@TEST.H5L.SE
2016-09-29T10:20:54 Looking for ENC-TS pa-data -- remove@TEST.H5L.SE
2016-09-29T10:20:54 Need to use PA-ENC-TIMESTAMP/PA-PK-AS-REQ
2016-09-29T10:20:54 sending 303 bytes to IPv6:::1
2016-09-29T10:20:54 reading packet: udp ::1:49188 (localhost) tid: 00000001
2016-09-29T10:20:54 host completed: udp ::1:49188 (localhost) tid: 00000001
2016-09-29T10:20:54 error message: Need to use PA-ENC-TIMESTAMP/PA-PK-AS-REQ: -1765328359
2016-09-29T10:20:54 krb5_sendto_context TEST.H5L.SE done: 0 hosts 1 packets 1 wc: 0.001588 nr: 0.000132 kh: 0.000028 tid: 00000002
2016-09-29T10:20:54 krb5_get_init_creds: loop 2
2016-09-29T10:20:54 krb5_get_init_creds: processing input
2016-09-29T10:20:54 krb5_get_init_creds: got an error
2016-09-29T10:20:54 error message: Need to use PA-ENC-TIMESTAMP/PA-PK-AS-REQ: -1765328359
2016-09-29T10:20:54 krb5_get_init_creds: KRB-ERROR -1765328359
2016-09-29T10:20:54 KDC send 7 patypes
2016-09-29T10:20:54 KDC send PA-DATA type: 16
2016-09-29T10:20:54 KDC send PA-DATA type: 15
2016-09-29T10:20:54 KDC send PA-DATA type: 147
2016-09-29T10:20:54 KDC send PA-DATA type: 2
2016-09-29T10:20:54 KDC send PA-DATA type: 138
2016-09-29T10:20:54 KDC send PA-DATA type: 136
2016-09-29T10:20:54 KDC send PA-DATA type: 19
2016-09-29T10:20:54 krb5_get_init_creds: using ENC-TS with enctype 18
2016-09-29T10:20:54 krb5_get_init_creds: using default_s2k_func
2016-09-29T10:20:54 error message: Did not find a plugin for send_to_kdc: 2
2016-09-29T10:20:54 Trying to find service kdc for realm TEST.H5L.SE flags 0
2016-09-29T10:20:54 error message: Did not find a plugin for service_locator: 2
2016-09-29T10:20:54 configuration file for realm TEST.H5L.SE found
2016-09-29T10:20:54 submissing new requests to new host
2016-09-29T10:20:54 error message: Did not find a plugin for send_to_kdc: 2
2016-09-29T10:20:54 connecting to host: udp ::1:49188 (localhost) tid: 00010001
2016-09-29T10:20:54 Queuing host in future (in 3s), its the 2 address on the same name: udp 127.0.0.1:49188 (localhost) tid: 00010002
2016-09-29T10:20:54 writing packet: udp ::1:49188 (localhost) tid: 00010001
2016-09-29T10:20:54 AS-REQ remove@TEST.H5L.SE from IPv6:::1 for krbtgt/TEST.H5L.SE@TEST.H5L.SE
2016-09-29T10:20:54 Client sent patypes: ENC-TS, REQ-ENC-PA-REP
2016-09-29T10:20:54 Looking for PK-INIT(ietf) pa-data -- remove@TEST.H5L.SE
2016-09-29T10:20:54 Looking for PK-INIT(win2k) pa-data -- remove@TEST.H5L.SE
2016-09-29T10:20:54 Looking for ENC-TS pa-data -- remove@TEST.H5L.SE
2016-09-29T10:20:54 ENC-TS Pre-authentication succeeded -- remove@TEST.H5L.SE using aes256-cts-hmac-sha1-96
2016-09-29T10:20:54 ENC-TS pre-authentication succeeded -- remove@TEST.H5L.SE
2016-09-29T10:20:54 error message: encryption type des-cbc-crc is disabled: -1765328234
2016-09-29T10:20:54 AS-REQ authtime: 2016-09-29T10:20:54 starttime: unset endtime: 2016-09-29T20:20:54 renew till: unset
2016-09-29T10:20:54 Client supported enctypes: aes256-cts-hmac-sha1-96, aes128-cts-hmac-sha1-96, des3-cbc-sha1, arcfour-hmac-md5, using aes256-cts-hmac-sha1-96/aes256-cts-hmac-
sha1-96
2016-09-29T10:20:54 Requested flags: forwardable
2016-09-29T10:20:54 sending 700 bytes to IPv6:::1
2016-09-29T10:20:54 reading packet: udp ::1:49188 (localhost) tid: 00010001
2016-09-29T10:20:54 host completed: udp ::1:49188 (localhost) tid: 00010001
2016-09-29T10:20:54 krb5_sendto_context TEST.H5L.SE done: 0 hosts 1 packets 1 wc: 0.001561 nr: 0.000105 kh: 0.000023 tid: 00010002
2016-09-29T10:20:54 krb5_get_init_creds: loop 3
2016-09-29T10:20:54 krb5_get_init_creds: processing input
2016-09-29T10:20:54 krb5_get_init_creds: using keyproc
2016-09-29T10:20:54 krb5_get_init_creds: using default_s2k_func
2016-09-29T10:20:54 krb5_get_init_creds: extracting ticket
2016-09-29T10:20:54 processing enc-ap-rep
2016-09-29T10:20:54 enc-ap-rep: valid
2016-09-29T10:20:54 error message: Did not find a plugin for ccache_ops: 2
2016-09-29T10:20:54 error message: open ../../tests/kdc/kdc.conf: No such file or directory: 2
2016-09-29T10:20:54 error message: failed to open ../../tests/kdc/mkey.file: No such file or directory: 2
signal killing kdc kpasswdd
FAIL check-kdc (exit status: 1)

FAIL: check-kdc-weak
====================

Creating database
lt-kadmin: kadm5_create_principal: Input/output error
lt-kadmin: adding krbtgt/SOME-REALM7.UK@SOME-REALM6.US: Input/output error
FAIL check-kdc-weak (exit status: 1)
```

**Comment by nicowilliams** [comment]
Thanks for linking that patch!

**Comment by jaltman** [comment]
should be fixed by 7c8b66d76b562912c09c0955a53da2f26afbc8f7 merged by @nicowilliams

**Comment by nicowilliams** [comment]
I've pushed 7c8b66d (based on the patch linked above, https://bugs.debian.org/cgi-bin/bugreport.cgi?att=1;bug=822749;filename=0001-Add-missing-type-cast-in-lib-kadm5-log.c.patch;msg=22) which should fix this.  I can't test on ARM though.
